tests/test_streaming.py

```diff
@@ -12,6 +12,103 @@
 from tests.common.utils import convert_to_mds
 
 
+@pytest.mark.parametrize('batch_size', [1, 4])
+@pytest.mark.parametrize('seed', [2222])
+@pytest.mark.parametrize('shuffle', [False])
+@pytest.mark.parametrize('drop_last', [False, True])
+@pytest.mark.parametrize('num_canonical_nodes', [1])
+@pytest.mark.parametrize('epoch_size', [10, 100])
+@pytest.mark.usefixtures('local_remote_dir')
+def test_dataloader_epoch_size_no_streams(local_remote_dir: Any, batch_size: int, seed: int,
+                                          shuffle: bool, drop_last: bool, num_canonical_nodes: int,
+                                          epoch_size: int):
+    remote_dir, local_dir = local_remote_dir
+    convert_to_mds(out_root=remote_dir,
+                   dataset_name='sequencedataset',
+                   num_samples=117,
+                   size_limit=1 << 8)
+
+    # Build StreamingDataset
+    dataset = StreamingDataset(local=local_dir,
+                               remote=remote_dir,
+                               shuffle=shuffle,
+                               batch_size=batch_size,
+                               shuffle_seed=seed,
+                               num_canonical_nodes=num_canonical_nodes,
+                               epoch_size=epoch_size)
+
+    # Build DataLoader
+    dataloader = StreamingDataLoader(dataset=dataset,
+                                     batch_size=batch_size,
+                                     num_workers=0,
+                                     drop_last=drop_last)
+
+    samples_seen = 0
+    for batch in dataloader:
+        samples_seen += batch['sample'].size(dim=0)
+
+    if drop_last:
+        assert samples_seen == epoch_size - (epoch_size % batch_size)
+    else:
+        assert samples_seen == epoch_size
+
+
+@pytest.mark.parametrize('batch_size', [1, 4])
+@pytest.mark.parametrize('seed', [2222])
+@pytest.mark.parametrize('shuffle', [False, True])
+@pytest.mark.parametrize('drop_last', [False])
+@pytest.mark.parametrize('num_canonical_nodes', [1])
```

Comment:
Can we have a variation in `num_canonical_nodes` ? such as `4` or `8` ? Plus increase the `num_samples` if it is necessary!